website/docs/language/functions/templatestring.mdx

```diff
@@ -0,0 +1,91 @@
+---
+sidebar_label: templatestring
+description: |-
+  The `templatestring` function takes a string and renders it as a template using a supplied set of template variables.
+---
+
+# `templatestring` Function
+
+The `templatestring` function takes a string and renders it as a template using a supplied set of template variables.
+
+```hcl
+templatestring(str, vars)
+```
+
+## Examples
+
+### Simple String Template
+
+A basic example showcasing the use of templatestring for a static string.
+
+```hcl
+output "result" {
+  value = templatestring("Hello, Jodie!", {})
+}
+```
+```
+result = Hello, Jodie!
+```
+### String interpolation with variable
+
+This example illustrates string interpolation by inserting a variable value into the template.
+
+```hcl
+output "result" {
+  value = templatestring("Hello, ${name}!", { name = "Alice" })
+}
+```
+```
+result = Hello, Alice!
+```
+## Advanced Usage
+
+Although in principle `templatestring` can be used with an inline template string, we don't recommend this approach because it requires awkward escaping. Instead, iconsider exploring alternative options such as you can use the local block to create a local variable and then pass that variable to the templatestring function or utilizing the file function. For a clearer understanding, refer to the example below.
```

Comment:
I don't think we should mention using the file function here, they should use the other template function there